command/agent/local.go

```diff
@@ -109,18 +109,18 @@ func (l *localState) ConsulServerUp() {
 // Pause is used to pause state synchronization, this can be
 // used to make batch changes
 func (l *localState) Pause() {
-	atomic.StoreInt32(&l.paused, 1)
+	atomic.AddInt32(&l.paused, 1)
 }
 
 // Resume is used to resume state synchronization
 func (l *localState) Resume() {
-	atomic.StoreInt32(&l.paused, 0)
+	atomic.AddInt32(&l.paused, -1)
```

Comment:
I was thinking about this as well. We could enforce an arbitrary upper limit on the semaphore to catch the case where someone is walking it up, which is probably the most likely programming error, but it won't catch an oddball bad path that doesn't get called often.